lib/id3c/api/datastore.py

```diff
@@ -382,6 +383,61 @@ def fetch_identifier_set_uses(session: DatabaseSession) -> Any:
 
         return list(cursor)
 
+def store_sample(session: DatabaseSession, sample: dict) -> Any:
+    """"
+    Validate the given *sample* and insert or update in the backing database.
+
+    Returns a list of in the same order as the input, with each object including
+    the ``sample_id`` (string), ``status`` (string) to indicate if inserted,
+    updated, or validation failed, and ``details`` to indicate reason for
+    failed validation.
+    """
+    with session:
```

Comment:
Won't this close the session when the context manager block exits? I'm curious about the interplay between that and `authenticated_datastore_session_required()` in lib/id3c/api/utils/routes.py, if any. It looks like the DatabaseSession `__exit__()` function calls the underlying psycopg2 connection `__exit__()` when the context manager is destroyed.

Reply:
Yes, the session should close when the request is completed. The session is passed in from the corresponding `post_sample` function in `lib/id3c/api/routes.py`. I mostly followed the pattern established for the other endpoints, but just realized that I left out the `@export` and `@catch_permission_denied` decorators on this function. I'll fix that now!